desc/plotting.py

```
@@ -1146,22 +1157,24 @@ def plot_fsa(  # noqa: C901
         fig, ax = plot_fsa(eq, "B_theta", with_sqrt_g=False)
 
     """
-    if np.isscalar(rho) and (int(rho) == rho):
-        rho = np.linspace(0, 1, rho + 1)
-    rho = np.atleast_1d(rho)
     if M is None:
         M = eq.M_grid
     if N is None:
         N = eq.N_grid
+    if grid is None:
```

might want to throw an error if grid and rho are both provided?

---

or at least document that grid overrides rho?